File: leetcode/0771-jewels-and-stones/solution.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        """
        RUNTIME: 25 ms (99.15%)
        MEMORY: 14 MB (12.16%)
        """
        hashmap = {}
        for i in stones:
            if i not in hashmap:
                hashmap[i] = 1
            else:
                hashmap[i] += 1
        total = 0
        for i in jewels:
            if i in hashmap:
                total += hashmap[i]
        return total

File: leetcode/0771-jewels-and-stones/solution.cpp
class Solution {
public:
    /**
     * RUNTIME: 0 ms (100%)
     * MEMORY: 6.3 MB (49.75%)
    */
    int numJewelsInStones(string jewels, string stones) {
        unordered_map <char, int> hashmap;
        int cnt = 0;
        for (char i : stones) {
            if (jewels.find(i) != string::npos) {
                ++hashmap[i];
            }
        }
        for (char i : jewels) {
            if (hashmap.find(i) != hashmap.end()) {
                cnt += hashmap[i];
            }
        }
        return cnt;
    }
};
